unity-client/Assets/Scripts/MainScripts/DCL/Components/TextShape/TextShape.cs

```diff
@@ -196,5 +180,38 @@ public static TextAlignmentOptions GetAlignment(string vTextAlign, string hTextA
                     }
             }
         }
+
+        public override void SetModel(object model)
+        {
+            this.model = (Model)model;
+            PrepareRectTransform();
+            ApplyCurrentModel();
+        }
```

Comment:
That sounds good to me but there is a big problem with that approach that we need to solve.

In builder in world, we create/update components client-side and send an update to the kernel that the component has changed. Some components will need some frames to apply the model that is the reason `ApplyChanges  ` returning an `IEnumerator ` and update its data. At the moment of creation of an entity, we tell kernel something like "New entity with X,Y,Z components and their values". In this situation we need components X,Y,Z to have their data updated in order to sent the correct data to Kernel.

However, since the components may need some frames to apply the changes, we could end in a situation where we are updating the components with old values. 

The correct way to handle this should be to separate the current data from the changes applies but since this is a big refactor and we don't know if this can affect heavily in the current system, we ended with this approach.